### PR TITLE
Docs power api and addheader 

### PR DESCRIPTION
**Pull request by Aincvy**
References #911

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

File: docs/src/main/paradox/server/walkthrough.md [hunk 1]
```diff
@@ -266,6 +267,8 @@ request metadata parameter to each service method. See the detailed chapters on
 and @ref[Maven](../buildtools/maven.md) for how to set this build option. Note that this option doesn't effect the
 generated client stubs.
 
+Notice: you need change `GreeterServiceHandlerFactory` to `GreeterServiceHandlerFactoryPowerApiHandlerFactory`
```
Comment (raboof) [inline docs/src/main/paradox/server/walkthrough.md:270]: I think these class names are java-specific. Could you wrap them in `@java[....]` and add the corresponding `@scala[...]` alternative?

File: docs/src/main/paradox/client/walkthrough.md [hunk 2]
```diff
@@ -171,3 +171,24 @@ Maven
 :   ```
 mvn akka-grpc:generate compile exec:java -Dexec.mainClass=io.grpc.examples.helloworld.GreeterClient
 ```
+
+
+
+## Request Metadata
```
[the remark this answers is not in the document]
Comment (Aincvy) [reply docs/src/main/paradox/client/walkthrough.md:177]: Keep walkthrough short is good.
But I don't why Accessing-request-metadata part of `Server` is in the walkthrough, and the `Client` side is in `details`. I think it should be in the same part.
Details or walkthrough, it's all ok.

File: docs/src/main/paradox/server/walkthrough.md (hunk 1 above)
Comment (Aincvy) [reply docs/src/main/paradox/server/walkthrough.md:270]: Sure.

File: docs/src/main/paradox/client/walkthrough.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (raboof) [reply docs/src/main/paradox/client/walkthrough.md:177]: That's a good point. Let's move it to 'details' (and add a link) for the server side as well.

File: docs/src/main/paradox/server/walkthrough.md (hunk 1 above)
Comment (Aincvy) [reply docs/src/main/paradox/server/walkthrough.md:270]: In scala, does the java class `GreeterServiceHandlerFactoryPowerApiHandlerFactory` is `GreeterServiceHandlerPowerApi`  ?

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

File: docs/src/main/paradox/server/details.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+By default the generated service interfaces don't provide access to the request metadata, only to the request
+body (via the RPC method input parameter). If your methods require access to the request @apidoc[Metadata], you can configure
+Akka gRPC to generate server "power APIs" that extend the base service interfaces to provide an additional
+request metadata parameter to each service method. See the detailed chapters on @ref[sbt](../buildtools/sbt.md), @ref[Gradle](../buildtools/gradle.md)
+and @ref[Maven](../buildtools/maven.md) for how to set this build option. Note that this option doesn't effect the
+generated client stubs.
+
+@java[Notice: you need change `GreeterServiceHandlerFactory` to `GreeterServiceHandlerFactoryPowerApiHandlerFactory`.]
+
+@scala[Notice: you need change `GreeterServiceHandler` to ``.]
```
Comment (raboof) [inline docs/src/main/paradox/server/details.md:14]:
```suggestion
@scala[Notice: you need change `GreeterServiceHandler` to `GreeterServicePowerApiHandler`.]
```

Comment (Aincvy) [reply docs/src/main/paradox/server/details.md:14]: You're welcome

**Comment by raboof** [comment]
@Aincvy I noticed some commits have `wangzhaoyuan <474551869@qq.com>` as the author, which is not recognized as someone who has signed the CLA. Is that you? Could you update those commits? (or do you want me to take care of it?)

**Comment by Aincvy** [comment]
Yeah, that's me. That was my first commit, and that is my company profile.  
Could you help me change it? I don't want to add the email to my GitHub account.
Thank you!  😊😊😊

**Comment by raboof** [comment]
@Aincvy could you sign the Lightbend CLA at https://www.lightbend.com/contribute/cla ?

**Comment by raboof** [comment]
Merged there!

**Pull request closed without merging**